### PR TITLE
Annihilate codepage & resolve startup segfault 

### PR DESCRIPTION
**Pull request by Headline**
I think this might have been the same case @KyleSanderson ran into. Not sure of it's cause, or why it has recently become a problem, but this should handle it just fine.

### CODE REVIEW

File: compiler/sc1.cpp [hunk 1]
```diff
@@ -1088,14 +1088,16 @@ static void setconfig(char *root)
 
       /* same for the codepage root */
 # if !defined NO_CODEPAGE
-        *ptr='\0';
+        if(ptr)
```
Comment (dvander) [inline compiler/sc1.cpp]: I don't see a way ptr can be NULL here, as it's NULL checked above:

`  if ((ptr=strrchr(path,DIRSEP_CHAR))!=NULL || (ptr=strchr(path,':'))!=NULL) {`

Comment (Headline) [reply compiler/sc1.cpp]: i think it’s reassigned and catches the plague here
https://github.com/alliedmodders/sourcepawn/blob/425c86a8f9d40c8bffec7b4bd2e6d32233c66f8f/compiler/sc1.cpp#L1077

Comment (Headline) [reply compiler/sc1.cpp]: for reference, here's my trace from the segfault
screen shot 2018-09-28 at 7 16 52 pm

**Comment by dvander** [comment]
Thanks!

**Pull request merged**